tests/cases/conformance/types/tuple/restTupleElements1.ts

```diff
@@ -69,3 +69,6 @@ declare function f2(...a: [(x: number) => number, ...((x: string) => number)[]])
 
 f1([x => x * 2, x => x.length, x => x.charCodeAt(0)]);
 f2(x => x * 2, x => x.length, x => x.charCodeAt(0));
+
+// @filename: /b.ts
+type T = [...string?];  // Error
```

Comment:
> Is this in a separate file because it's now a parser error and that blocks the checker errors in the other file?

So. Do you think it would be better to move this handling to the `checker`? What is the best way to check `!`/`?` token in the checker? Can we add new prop to `JSDocNonNullableType`/`JSDocNullableType`, similar to `postfix`?

Reply:
> Is this in a separate file because it's now a parser error and that blocks the checker errors in the other file?

So. Do you think it would be better to move this handling to the `checker`?